clang/unittests/Analysis/FlowSensitive/DataflowEnvironmentTest.cpp

```diff
@@ -442,6 +442,46 @@ TEST_F(EnvironmentTest, CXXDefaultInitExprResultObjIsWrappedExprResultObj) {
             &Env.getResultObjectLocation(*DefaultInit->getExpr()));
 }
 
+TEST_F(EnvironmentTest, ResultObjectLocationForInheritedCtorInitExpr) {
```

Comment:
We tend to put tests for `getResultObjectLocation()` in TransferTest. See an example for `AtomicExpr` [here](https://github.com/llvm/llvm-project/blob/56ad7cc0126f9899fd884391cfa10b6359206c01/clang/unittests/Analysis/FlowSensitive/TransferTest.cpp#L3391), which you can also use as a model for a test for `CXXInheritedCtorInitExpr`.

Doing this in TransferTest.cpp has the benefit that the test is smaller, plus you can make the assertion stronger (see the test for `AtomicExpr` linked above which checks the exact value returned by `getResultObjectLocation()` instead of just testing that it is non-null).

Reply:
Thanks for looking at this Martin!

I tried doing that, but in that way I can't have access to the right `Environment`.
The problem is that the specific part of the AST I am interested to (a `CXXConstructorDecl`) is implicitly "generated" (because it is an inherited constructor), so I can't put an annotation inside of it. 
I may be completely wrong, but I think that in this case the only way is to explicitly initialize the environment as it is usually done in EnvironmentTest.

Reply:
> The problem is that the specific part of the AST I am interested to (a CXXConstructorDecl) is implicitly "generated" (because it is an inherited constructor), so I can't put an annotation inside of it.

Ah, thanks for the explanation, I understand now. Yes, agree that you can't do this in the way we do other tests for `getResultObjectLocation()` in TransferTest.cpp.

Can you add a comment to the test that explains this (and points out that most of the tests for `getResultObjectLocation()` are in TransferTest.cpp)? I'd like to avoid other tests for `getResultObjectLocation()` getting added next to this test just because it looks like the right place for them.